build.gradle.kts

```diff
@@ -0,0 +1,268 @@
+        isVisible = false
+    }
+    "coreRuntimeExtensions" {
+        attributes { attribute(Usage.USAGE_ATTRIBUTE, project.objects.named(Usage::class.java, Usage.JAVA_RUNTIME)) }
+        isCanBeResolved = true
+        isCanBeConsumed = false
+        isVisible = false
+    }
+    "externalModules" {
+        isVisible = false
+    }
+    // Configuration used to resolve external modules before patching them with versions from core runtime
+    "externalModulesRuntime" {
+        isVisible = false
+        extendsFrom(configurations["coreRuntime"])
+        extendsFrom(configurations["externalModules"])
+    }
+    // Combines the 'coreRuntime' with the patched external module jars
+    "runtime" {
+        isVisible = false
+        extendsFrom(configurations["coreRuntime"])
+    }
+    "gradlePlugins" {
+        isVisible = false
+    }
+    "testRuntime" {
+        extendsFrom(configurations["runtime"])
+        extendsFrom(configurations["gradlePlugins"])
+    }
+
+    all {
+        attributes.attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage::class.java, Usage.JAVA_RUNTIME))
+    }
+}
+
+extra.set("allTestRuntimeDependencies", configurations["testRuntime"].allDependencies)
+
+val patchedExternalModulesDir = File(buildDir, "external/files")
+val patchedExternalModules = files(Callable { fileTree(patchedExternalModulesDir).files.sorted() })
+patchedExternalModules.builtBy("patchExternalModules")
+
+dependencies {
+    "externalModules"("org.gradle:gradle-kotlin-dsl:${BuildEnvironment.gradleKotlinDslVersion}")
+    "externalModules"("org.gradle:gradle-kotlin-dsl-tooling-builders:${BuildEnvironment.gradleKotlinDslVersion}")
+    "coreRuntime"(project(":launcher"))
+    "coreRuntime"(project(":runtimeApiInfo"))
+    "runtime"(project(":wrapper"))
+    "runtime"(project(":installationBeacon"))
+    "runtime"(patchedExternalModules)
+    pluginProjects.forEach { "gradlePlugins"(it) }
+    implementationPluginProjects.forEach { "gradlePlugins"(it) }
+    "gradlePlugins"(project(":workers"))
+    "gradlePlugins"(project(":dependencyManagement"))
+    "gradlePlugins"(project(":testKit"))
+
+    "coreRuntimeExtensions"(project(":dependencyManagement")) //See: DynamicModulesClassPathProvider.GRADLE_EXTENSION_MODULES
+    "coreRuntimeExtensions"(project(":pluginUse"))
+    "coreRuntimeExtensions"(project(":workers"))
+    "coreRuntimeExtensions"(patchedExternalModules)
+}
+
+extra.set("allCoreRuntimeExtensions", configurations["coreRuntimeExtensions"].allDependencies)
+
+tasks.create<PatchExternalModules>("patchExternalModules") {
+    allModules = configurations["externalModulesRuntime"]
+    coreModules = configurations["coreRuntime"]
+    modulesToPatch = configurations["externalModules"]
+    destination = patchedExternalModulesDir
+}
+
+evaluationDependsOn(":distributions")
+
+tasks.create<Install>("install") {
+    description = "Installs the minimal distribution into directory ${project.property("gradle_installPath")}"
```

Comment:
:x: `$gradle_installPath` should not be evaluated in this description
👍 to @blindpirate's [comment](https://github.com/gradle/gradle/pull/4851/files?diff=unified#r178696438).
Note that to escape `$` in a Kotlin string you can use `${"$"}`

Also applies to the `installAll` task below